prog/types.go

```diff
@@ -569,6 +576,10 @@ func (t *ArrayType) isDefaultArg(arg Arg) bool {
 	return true
 }
 
+func (t *ArrayType) Alignment() uint64 {
```

Comment:
If we have TypeCommon.TypeAlign, I think we don't need this. TypeCommon.Alignment should return the correct value for ArrayType as well.